subprojects/base-services/src/main/java/org/gradle/internal/progress/BuildOperationDescriptor.java

```diff
@@ -21,21 +21,27 @@
 /**
  * Meta-data about a build operation.
  */
-public class BuildOperationDetails {
-    private final BuildOperationExecutor.Operation parent;
+public class BuildOperationDescriptor {
+    private final Long id;
+    private final Long parentId;
     private final String displayName;
     private final String name;
     private final String progressDisplayName;
     private final Object operationDescriptor;
 
-    private BuildOperationDetails(BuildOperationExecutor.Operation parent, String name, String displayName, String progressDisplayName, Object operationDescriptor) {
-        this.parent = parent;
+    protected BuildOperationDescriptor(Long id, Long parentId, String name, String displayName, String progressDisplayName, Object operationDescriptor) {
+        this.id = id;
+        this.parentId = parentId;
         this.name = name;
         this.displayName = displayName;
         this.progressDisplayName = progressDisplayName;
         this.operationDescriptor = operationDescriptor;
```

Comment:
The build scan plugin does not use this class. It uses `BuildOperationInternal`.